src/main/scala/edg_ide/runner/PDFGeneratorUtil.scala

```diff
@@ -48,22 +102,54 @@ object PDFGeneratorUtil{
         }
 
         val cb = writer.getDirectContent
-        val graphics = cb.createGraphics(width, height)
+        val graphics = cb.createGraphics(width, adjustedHeight)
         val painter = new ElkNodePainter(node)
         painter.paintComponent(graphics, Color.white)
         graphics.dispose()
+
+        // Prints out table if there are multiple usages of the same component
+        if(dupSet.size > 1) {
+          val table = new PdfPTable(1)
+
+          val message = new Paragraph(s"${className.toSimpleString} is also used at the following:")
+          val title = new PdfPCell(message)
+          title.setBorder(Rectangle.NO_BORDER)
+          table.addCell(title)
+
+          dupSet.foreach { path =>
+            val parentAnchor = new Anchor
+            val (parent, current) = path.split
+            parentAnchor.setReference(s"#${parent.toString}")
+            parentAnchor.add(new Phrase(s"${parent.toString}", font))
+            parentAnchor.add(new Phrase(s".${current.toString}"))
+
+            val cellContent = new Paragraph
+            cellContent.add(parentAnchor)
+            val cell = new PdfPCell(cellContent)
+            cell.setFixedHeight(TABLE_ROW_HEIGHT)
+            table.addCell(cell)
+          }
+          val tableY = adjustedHeight - height + ElkNodePainter.margin
+          table.setTotalWidth(width - 2 * ElkNodePainter.margin)
+          table.writeSelectedRows(0, -1, ElkNodePainter.margin.toFloat, tableY, writer.getDirectContent())
+        }
       }
 
+
       def printNextHierarchyLevel(block: HierarchyBlock, path: DesignPath = DesignPath()): Unit = {
-        val node = HierarchyGraphElk.HBlockToElkNode(block, path, mappers=mappers)
-        printNode(node)
+        val node = HierarchyGraphElk.HBlockToElkNode(block, path, mappers = mappers)
+        printNode(node, block.getSelfClass, path)
 
         block.blocks.asPairs.map {
           case (name, subblock) => (name, subblock.`type`)
         }.collect {
-          case (name, BlockLike.Type.Hierarchy(subblock)) if subblock.blocks.nonEmpty => (path + name, subblock)
+          case (name, BlockLike.Type.Hierarchy(subblock)) if subblock.blocks.nonEmpty => (path + name, subblock, subblock.getSelfClass)
         }.foreach {
-          case (path, subblock) => printNextHierarchyLevel(subblock, path)
+          case (path, subblock, className) => {
+            if (path == dupList.getOrElse(className, Set.empty).head) {
```

Comment:
```suggestion
            if (dupList.getOrElse(className, Set.empty).headOption.contains(path)) {
```

I think Set.empty.head() will crash, though in practice it seems structurally it also shouldn't encounter dupList not having the entry. headOption returns None instead of crashing.